tests/spin/test_spin_hamiltonian.py

```diff
@@ -782,3 +783,826 @@ def test_fermi_hubbard_hamiltonian_matrix(shape, n_cells, t, coulomb, expected_h
+            lattice=lattice,
+            n_cells=n_cells,
+            hopping=[1.0],
+            intersite_coupling=[1.0, 2.0],
+            neighbour_order=1,
+        )
+
+
+def test_mapping_error_emery():
+    r"""Test that an error is raised when unsupported mapping is provided"""
+    n_cells = [4, 4]
+    lattice = "Square"
+    with pytest.raises(ValueError, match="The 'bk_sf' transformation is not available."):
+        emery(lattice=lattice, n_cells=n_cells, mapping="bk_sf")
+
+
+@pytest.mark.parametrize(
+    # expected_ham here was obtained manually
+    ("shape", "n_cells", "t", "u", "v", "boundary_condition", "expected_ham"),
+    [
+        (
+            "square",
+            [2, 2],
+            -1.23,
+            2.34,
+            1.42,
+            False,
+            (0.615 + 0j) * (Y(0) @ Z(1) @ Y(2))
+            + (0.615 + 0j) * (X(0) @ Z(1) @ X(2))
+            + (0.615 + 0j) * (Y(1) @ Z(2) @ Y(3))
+            + (0.615 + 0j) * (X(1) @ Z(2) @ X(3))
+            + (0.615 + 0j) * (Y(0) @ Z(1) @ Z(2) @ Z(3) @ Y(4))
+            + (0.615 + 0j) * (X(0) @ Z(1) @ Z(2) @ Z(3) @ X(4))
+            + (0.615 + 0j) * (Y(1) @ Z(2) @ Z(3) @ Z(4) @ Y(5))
+            + (0.615 + 0j) * (X(1) @ Z(2) @ Z(3) @ Z(4) @ X(5))
+            + (0.615 + 0j) * (Y(2) @ Z(3) @ Z(4) @ Z(5) @ Y(6))
+            + (0.615 + 0j) * (X(2) @ Z(3) @ Z(4) @ Z(5) @ X(6))
+            + (0.615 + 0j) * (Y(3) @ Z(4) @ Z(5) @ Z(6) @ Y(7))
+            + (0.615 + 0j) * (X(3) @ Z(4) @ Z(5) @ Z(6) @ X(7))
+            + (0.615 + 0j) * (Y(4) @ Z(5) @ Y(6))
+            + (0.615 + 0j) * (X(4) @ Z(5) @ X(6))
+            + (0.615 + 0j) * (Y(5) @ Z(6) @ Y(7))
+            + (0.615 + 0j) * (X(5) @ Z(6) @ X(7))
+            + (8.020000000000005 + 0j) * I(0)
+            + (-2.005 + 0j) * Z(1)
+            + (-2.005 + 0j) * Z(0)
+            + (0.585 + 0j) * (Z(0) @ Z(1))
+            + (-2.005 + 0j) * Z(3)
+            + (-2.005 + 0j) * Z(2)
+            + (0.585 + 0j) * (Z(2) @ Z(3))
+            + (-2.005 + 0j) * Z(5)
+            + (-2.005 + 0j) * Z(4)
+            + (0.585 + 0j) * (Z(4) @ Z(5))
+            + (-2.005 + 0j) * Z(7)
+            + (-2.005 + 0j) * Z(6)
+            + (0.585 + 0j) * (Z(6) @ Z(7))
+            + (0.355 + 0j) * (Z(0) @ Z(2))
+            + (0.355 + 0j) * (Z(0) @ Z(3))
+            + (0.355 + 0j) * (Z(1) @ Z(2))
+            + (0.355 + 0j) * (Z(1) @ Z(3))
+            + (0.355 + 0j) * (Z(0) @ Z(4))
+            + (0.355 + 0j) * (Z(0) @ Z(5))
+            + (0.355 + 0j) * (Z(1) @ Z(4))
+            + (0.355 + 0j) * (Z(1) @ Z(5))
+            + (0.355 + 0j) * (Z(2) @ Z(6))
+            + (0.355 + 0j) * (Z(2) @ Z(7))
+            + (0.355 + 0j) * (Z(3) @ Z(6))
+            + (0.355 + 0j) * (Z(3) @ Z(7))
+            + (0.355 + 0j) * (Z(4) @ Z(6))
+            + (0.355 + 0j) * (Z(4) @ Z(7))
+            + (0.355 + 0j) * (Z(5) @ Z(6))
+            + (0.355 + 0j) * (Z(5) @ Z(7)),
+        ),
+        (
+            "Square",
+            [3, 3],
```

Comment:
This test seems relatively repetitive with the previous one. Consider keeping the reference Hamiltonians tests compact but comprehensive.

Reply:
This tests boundary_conditions being set to True/False.

Reply:
Then you can stay with a smaller cell to reduce the size of the reference Hamiltonian.

Reply:
This is the smallest square that I can generate where the boundary conditions make a difference. I have changed it to a chain for simpler reference Hamiltonian.